### PR TITLE
Feature/crud document node

### PR DESCRIPTION
**Pull request by GhenadieCebanu**
CRUD for document node
Issue: https://github.com/code4romania/expert-consultation-api/issues/4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -123,6 +124,12 @@
 			<artifactId>springfox-swagger-ui</artifactId>
 			<version>2.9.2</version>
 		</dependency>
+		<!-- MapStruct -->
+		<dependency>
```
Comment (georgebejan) [inline pom.xml:128]: we already have `modelmapper` for conversion

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    private final DocumentNodeMapper mapper;
+
+    @ApiOperation(value = "Return document node based on id",
+            response = DocumentNodeDto.class,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public DocumentNodeDto getDocumentById(@ApiParam("Id of the document node object being requested") @PathVariable UUID id) {
+        DocumentNode documentNode = documentNodeService.findById(id);
+
+        return mapper.map(documentNode);
```
Comment (georgebejan) [inline src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:33]: use the model mapper, check other services to see how it's done

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+public class DocumentNodeController {
+
+    private final DocumentNodeService documentNodeService;
+    private final DocumentNodeMapper mapper;
+
+    @ApiOperation(value = "Return document node based on id",
+            response = DocumentNodeDto.class,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public DocumentNodeDto getDocumentById(@ApiParam("Id of the document node object being requested") @PathVariable UUID id) {
```
Comment (georgebejan) [inline src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:30]: the name of the method should be `getDocumentNodeById` or just `getDocumentNode`

Comment (GhenadieCebanu) [reply src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java]: renamed

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+        return mapper.map(documentNode);
+    }
+
+    @ApiOperation(value = "Create a new document node in the platform",
+            response = UUID.class,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<UUID> create(
```
Comment (mihairinzis) [inline src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:41]: please also add tests for the new stuff

Comment (GhenadieCebanu) [reply src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:42]: sure, cumming in next commit

File: pom.xml (hunk 1 above)
Comment (GhenadieCebanu) [reply pom.xml:128]: there will be another task to replace modelmapper with MapStruct

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java (hunk 2 above)
Comment (GhenadieCebanu) [reply src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:34]: there will be another task to replace modelmapper with MapStruct

File: src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java (hunk 4 above)
Comment (GhenadieCebanu) [reply src/main/java/com/code4ro/legalconsultation/controller/DocumentNodeController.java:42]: added

**Comment by RaduCStefanescu** [comment]
Thank you for your work @GhenadieCebanu and @georgebejan

**Pull request merged**